apps/meteor/server/methods/saveUserPreferences.ts

```diff
@@ -169,6 +172,10 @@ export const saveUserPreferences = async (settings: Partial<UserPreferences>, us
 		if (Array.isArray(settings.highlights)) {
 			await Subscriptions.updateUserHighlights(user._id, settings.highlights);
 		}
+
+		if (settings.language && oldLanguage !== settings.language && rcSettings.get('AutoTranslate_AutoEnableOnJoinRoom')) {
+			await Subscriptions.updateAllAutoTranslateLanguagesByUserId(user._id, settings.language);
+		}
```

Comment:
About this one, can we try to update other methods on this setImmediate to use a single query when possible?

This callback does a bunch of heavy ops that could potentially update a ton of records, so if we can make them go on the fewest possible calls, that would be ideal.

You can do those changes with some tests after this one is merged and before the release is cut, wdyt?

Reply:
Yea, that sounds good